python/sglang/srt/model_executor/model_runner.py

```diff
@@ -148,6 +148,11 @@ def load_model(self):
             f"[gpu={self.gpu_id}] Load weight begin. "
             f"avail mem={get_available_gpu_memory(self.gpu_id):.2f} GB"
         )
+        if torch.cuda.get_device_capability()[0] < 8:
+            logger.info(
+                "Compute capability below sm80 use float16 due to lack of bfloat16 support."
+            )
+            self.server_args.dtype = "float16"
```

Comment:
Considering that LLM inference dtype hardly uses float32 (float), this change is safe.
https://github.com/sgl-project/sglang/blob/5d0d40d0eb8c347d8b3598f0a375696728df66c4/python/sglang/srt/server_args.py#L183-L196